### PR TITLE
refactor: card primitive

### PR DESCRIPTION
**Pull request by alexcarpenter**
🔍 [Preview Link](https://react-components-git-accard-refactor-hashicorp.vercel.app/components/card)

---

## Description

Refactors Card to remove the optional children or props approach and exports card primitives to be used to build up our different card variant needs.

The base card variant should be unaffected in use on dotcom and within RelatedContent.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/card/docs.mdx [hunk 1]
```diff
@@ -215,72 +122,34 @@ A component used to promote and link to marketing pages, always using a descript
...
-
-<PropsTable props={componentProps.metaProps} />
-
-### `<Heading />`
-
-<PropsTable props={componentProps.headingProps} />
-
-### `<Description />`
-
-<PropsTable props={componentProps.descriptionProps} />
```
Comment (EnMod) [inline packages/card/docs.mdx:286]: Should these prop docs be removed?

Comment (alexcarpenter) [reply packages/card/docs.mdx:286]: They were, I think your commenting on code that was deleted.

Comment (EnMod) [reply packages/card/docs.mdx:286]: Right, I'm asking if they _should_ be removed :+1: I was thinking we should have prop docs in this file for the component?

Comment (alexcarpenter) [reply packages/card/docs.mdx:286]: ah I see what you mean, I can bring them back if you'd like. I though folks would not be interacting with these sub components outside of building the variants within react-components.

Comment (alexcarpenter) [reply packages/card/docs.mdx:286]: ah yeah, good call! https://github.com/hashicorp/react-components/pull/810/commits/4deaaaf47bc92c5b6f282ea1be4877f14a2c36cf

**Review by EnMod — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**